### PR TITLE
Chapter 1 - Birth

### PR DESCRIPTION
**Pull request by kasparm**
In this PR we will work on implementing the following concepts into the YegorCallCenter code:
- [x] Never use -er names for classes - [Blog post](https://www.yegor256.com/2015/03/09/objects-end-with-er.html)
- [x] Make one constructor primary - [Blog post](https://www.yegor256.com/2015/05/28/one-primary-constructor.html)
- [x] Keep constructor code free - [Blog post](https://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html)

### CODE REVIEW

**Comment by kasparm** [comment]
The main goal we are trying to achieve is writing code that is maintainable. The concepts discussed by Yegor have that purpose in mind.

- [x] **The Never use -er names for classes** section discusses that we should treat classes and objects as a thing and not a process. Yegor mentions that objects are a representation of its encapsulated data and not a collection of procedures.
Is I can tell from the classes in the YegorCallCenter, the classes are labeled according to that concept even though there are object names that end in -er but these are exceptions of that -er rule.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
+
+## Purpose
+Yegor Bugayenko wrote a book about opject-oriented programing callend Elegand Objects. Some people call the approach contravertial, some are trying to adapt the concepts from the book and some whould like to propose changes.
```
Comment (rouille) [inline README.md]: callend --> called

File: README.md [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+- [ ] Never us -er names
+- [ ] Make one constructor primary
+- [ ] Keep constructor code free
+
 # Call Center Design
+
 Source: https://github.com/donnemartin/system-design-primer/blob/master/solutions/object_oriented_design/call_center/call_center.ipynb
+
+## Design
+We will designing this with 4 classes:
```
Comment (rouille) [inline README.md]: We will designing --> We will be designing

File: README.md [hunk 3]
```diff
@@ -1,5 +1,30 @@
+
+## Purpose
+Yegor Bugayenko wrote a book about opject-oriented programing called Elegand Objects. Some people call the approach contravertial, some are trying to adapt the concepts from the book and some whould like to propose changes.
```
Comment (jenhagg) [inline README.md]: Elegand -> Elegant

Comment (jenhagg) [reply README.md]: contravertial -> contraversial

**Comment by kasparm** [comment]
@rouille @jenhagg thanks for the text edits... installed a spell check on my IDE to avoid that in the future :)

**Comment by kasparm** [comment]
- [x] **Make one constructor primary**
The concept here is to have several constructors and one primary one. Yegor's point is that a cohesive and robust class has a small number of methods and a large number of constructors. This concept will be discussed further in later PR's. The more constructors you provide, the more flexibility you provide to the user. The main job of a constructor is to initialize encapsulated properties using the provided arguments. Yegor recommends to put the initialization into one constructor and call this primary constructor by all other secondary constructors.

The existing code only has one constructor per class which is good enough for this example. I will just do minor chances and provide default argument values. Since python does not support function overloading directly, we could choose one of the available techniques to allow multiple constructors. [Here](https://realpython.com/python-multiple-constructors/#defining-multiple-class-constructors) is a list of techniques.

**Comment by rouille** [comment]
Does the `Caller` class be renamed to satisfy the first concept, i.e., `Never use -er names for classes`?

**Comment by kasparm** [comment]
Good point @rouille. I think in this case this is an exception because the `Caller` class represents the person that calls the call center. The name does not stand for an operation the class is performing.

**Comment by kasparm** [comment]
- [x] **Keep constructor code free**
Since we don't know what will happen with this code later and we may want to refactor it at some point, it is good to keep the constructor code free. This helps making the code more maintainable. The only action we want the constructor to do is to create the object.

**Review by rouille — APPROVED** [review]
LGTM

**Pull request merged**